extensions-support/bouncycastle/runtime/src/main/java/org/apache/camel/quarkus/support/bouncycastle/BouncyCastleRecorder.java

```diff
@@ -35,11 +34,19 @@ public class BouncyCastleRecorder {
 
     public void registerBouncyCastleProvider(List<String> cipherTransformations, ShutdownContext shutdownContext) {
         Provider provider = Security.getProvider(SecurityProviderUtils.BOUNCYCASTLE_PROVIDER_NAME);
+        if (provider == null) {
+            provider = Security.getProvider(SecurityProviderUtils.BOUNCYCASTLE_FIPS_PROVIDER_NAME);
+        }
         if (provider == null) {
             // TODO: Fix BuildStep execution order so that this is not required
             // https://github.com/apache/camel-quarkus/issues/3472
-            provider = new BouncyCastleProvider();
-            Security.addProvider(provider);
+            try {
+                provider = (Provider) Thread.currentThread().getContextClassLoader()
+                        .loadClass("org.bouncycastle.jce.provider.BouncyCastleProvider").getConstructor().newInstance();
+                Security.addProvider(provider);
+            } catch (Exception e) {
+                // Might happen if bouncy castle in not present (in FIPS system)
```

Comment:
Then provider stays <code>null</code> on fips system ? In other words, it throws on new <code>Instance()</code> or <code>addProvider()</code> ?

Reply:
added condition for loading fips provider